### PR TITLE
Animation: Clean up

### PR DESCRIPTION
**Pull request by Mugen87**
Some clean up in animation entities...
- added missing semicolons
- removed needless variable init with `undefined`
- adjust compare operation

### CODE REVIEW

**Comment by mrdoob** [comment]
Thanks!

**Comment by DelvarWorld** [comment]
I would appreciate if these files were documented - or should general consumers of threejs not be using them yet? so far they've been in at least 2 releases of three without any documentation

**Comment by Mugen87** [comment]
@mrdoob One more thing. What do you think about this [line](https://github.com/mrdoob/three.js/blob/dev/src/animation/AnimationMixer.js#L775)? Looks like there is missing a `break` statement...

**Comment by tschw** [comment]
Awe, so many typos... Thanks for fixing!

@Mugen87 
> One more thing. What do you think about this line? Looks like there is missing a break statement...

**No! It's an intentional fall-through case!** I often use a `// -V-V-V-` comment to denote it, but must've forgotten it, here.

```javascript
case X:
    // [...]
    break;
case X:
    // [...]
    // -V-V-V-
case Z:
```

**Comment by tschw** [comment]
Strictly following the style guide, [this `break`](https://github.com/mrdoob/three.js/blob/dev/src/animation/AnimationMixer.js#L861) would also need to be removed.

**Comment by mrdoob** [comment]
> @mrdoob One more thing. What do you think about this [line](https://github.com/mrdoob/three.js/blob/dev/src/animation/AnimationMixer.js#L775)? Looks like there is missing a `break` statement...

That's hard to follow code indeed.
I would just do `pingPong = ( loop === THREE.LoopPingPong );` when defining `pingPong` before the switch.

**Comment by tschw** [comment]
@Mugen87 
> Looks like there is missing a break statement...

@mrdoob  
> I would just do pingPong = ( loop === THREE.LoopPingPong ); when defining pingPong before the switch.

Express it in whatever form you like, just don't `break` it :stuck_out_tongue_winking_eye: 

It's probably best done without `switch` at all:
```javascript
if ( loop === THREE.LoopOnce ) {

    // [... code of the once case goes here ]

} else {

    var pingPong = ( loop === THREE.LoopPingPong );
    // [... common code for the repetitive cases (repeat / ping-pong) here ]

}
```

**Comment by mrdoob** [comment]
Yep! Also, remember that V8 can't optimize when there is a switch.

**Comment by tschw** [comment]
OT chat:

@mrdoob 
> Yep! Also, remember that V8 can't optimize when there is a switch.

I think it can't when you switch on stuff other than integers.

Not sure it tolerates numbers living in properties (not a problem with Google's own code - Closure advanced mode bakes `@const` properties to raw numbers). Maybe have to `freeze` the object...

Generally speaking, `switch` is potentially more optimal than `if`, especially when there are many cases; it hints the compiler to construct a decision tree, so only log(N) comparisons are needed instead of N. Traditional `switch` (in C), only allows constant integers known at compile time for the cases (in C that includes `enum`s which are named integer constants).

**Comment by MasterJames** [comment]
I imagine object key lookup being more of a best match one day then you could just put w for width etc.
